From: Byron Campen <docfaraday@gmail.com>
Date: Mon, 29 Jul 2024 12:06:50 -0500
Subject: (tmp-cherry-pick) Revert "Run IWYU on some files I intend to work on"
 (fe34363ca0)

This reverts commit 827da15f1408a399ed15ce5c9726b6af772fb71a.

Reason for revert: Breaks downstream project

Original change's description:
> Run IWYU on some files I intend to work on
>
> and files that broke when I fixed the first set.
>
> Bug: webrtc:42226242
> Change-Id: I321cd63537ab3002098c7bdecd889a6fc5a1eb25
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/353421
> Commit-Queue: Harald Alvestrand <hta@webrtc.org>
> Commit-Queue: Mirko Bonadei <mbonadei@webrtc.org>
> Reviewed-by: Mirko Bonadei <mbonadei@webrtc.org>
> Auto-Submit: Harald Alvestrand <hta@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#42429}

Bug: webrtc:42226242
Change-Id: I6b18dced08669c6741c6a51768fbb8b9072c6e82
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/353580
Owners-Override: Mirko Bonadei <mbonadei@webrtc.org>
Commit-Queue: Mirko Bonadei <mbonadei@webrtc.org>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/heads/main@{#42430}
---
 api/BUILD.gn                                  |  5 ---
 api/frame_transformer_interface.h             | 10 ++---
 api/rtp_receiver_interface.cc                 | 10 -----
 audio/BUILD.gn                                |  6 ---
 ...nnel_receive_frame_transformer_delegate.cc |  9 ----
 ...annel_receive_frame_transformer_delegate.h |  1 -
 ...ive_frame_transformer_delegate_unittest.cc |  7 ----
 ...end_frame_transformer_delegate_unittest.cc |  7 ----
 media/BUILD.gn                                |  1 -
 media/base/codec.cc                           | 10 -----
 media/base/codec.h                            |  4 +-
 modules/rtp_rtcp/BUILD.gn                     |  1 -
 ...sender_video_frame_transformer_delegate.cc | 20 +--------
 ..._sender_video_frame_transformer_delegate.h |  2 -
 pc/BUILD.gn                                   | 13 ------
 pc/rtp_receiver.cc                            |  3 --
 pc/rtp_receiver.h                             |  9 ++++
 pc/rtp_receiver_proxy.h                       |  9 ----
 pc/rtp_sender_receiver_unittest.cc            |  5 +--
 pc/rtp_transceiver.cc                         | 31 ++------------
 pc/sdp_offer_answer.cc                        | 42 ++-----------------
 pc/sdp_offer_answer.h                         |  1 +
 22 files changed, 26 insertions(+), 180 deletions(-)

diff --git a/api/BUILD.gn b/api/BUILD.gn
index b0f95ebcf8..f25db96db8 100644
--- a/api/BUILD.gn
+++ b/api/BUILD.gn
@@ -393,16 +393,11 @@ rtc_source_set("frame_transformer_interface") {
     "frame_transformer_interface.h",
   ]
   deps = [
-    ":array_view",
     ":make_ref_counted",
     ":ref_count",
     ":scoped_refptr",
-    "../rtc_base:refcount",
-    "../rtc_base/system:rtc_export",
-    "units:timestamp",
     "video:encoded_frame",
     "video:video_frame_metadata",
-    "//third_party/abseil-cpp/absl/types:optional",
   ]
 }
 
diff --git a/api/frame_transformer_interface.h b/api/frame_transformer_interface.h
index 6f632bcb72..03f7b482b8 100644
--- a/api/frame_transformer_interface.h
+++ b/api/frame_transformer_interface.h
@@ -11,17 +11,13 @@
 #ifndef API_FRAME_TRANSFORMER_INTERFACE_H_
 #define API_FRAME_TRANSFORMER_INTERFACE_H_
 
-#include <cstdint>
 #include <memory>
 #include <string>
 
-#include "absl/types/optional.h"
-#include "api/array_view.h"
 #include "api/ref_count.h"
 #include "api/scoped_refptr.h"
-#include "api/units/timestamp.h"
+#include "api/video/encoded_frame.h"
 #include "api/video/video_frame_metadata.h"
-#include "rtc_base/system/rtc_export.h"
 
 namespace webrtc {
 
@@ -108,7 +104,7 @@ class TransformableAudioFrameInterface : public TransformableFrameInterface {
 };
 
 // Objects implement this interface to be notified with the transformed frame.
-class TransformedFrameCallback : public RefCountInterface {
+class TransformedFrameCallback : public rtc::RefCountInterface {
  public:
   virtual void OnTransformedFrame(
       std::unique_ptr<TransformableFrameInterface> frame) = 0;
@@ -125,7 +121,7 @@ class TransformedFrameCallback : public RefCountInterface {
 
 // Transforms encoded frames. The transformed frame is sent in a callback using
 // the TransformedFrameCallback interface (see above).
-class FrameTransformerInterface : public RefCountInterface {
+class FrameTransformerInterface : public rtc::RefCountInterface {
  public:
   // Transforms `frame` using the implementing class' processing logic.
   virtual void Transform(
diff --git a/api/rtp_receiver_interface.cc b/api/rtp_receiver_interface.cc
index 4868f2c3aa..ffa421683a 100644
--- a/api/rtp_receiver_interface.cc
+++ b/api/rtp_receiver_interface.cc
@@ -10,16 +10,6 @@
 
 #include "api/rtp_receiver_interface.h"
 
-#include <string>
-#include <vector>
-
-#include "api/crypto/frame_decryptor_interface.h"
-#include "api/dtls_transport_interface.h"
-#include "api/frame_transformer_interface.h"
-#include "api/media_stream_interface.h"
-#include "api/scoped_refptr.h"
-#include "api/transport/rtp/rtp_source.h"
-
 namespace webrtc {
 
 std::vector<std::string> RtpReceiverInterface::stream_ids() const {
diff --git a/audio/BUILD.gn b/audio/BUILD.gn
index fb9e904d83..688a7adbc6 100644
--- a/audio/BUILD.gn
+++ b/audio/BUILD.gn
@@ -161,16 +161,12 @@ if (rtc_include_tests) {
       ":audio",
       ":audio_end_to_end_test",
       ":channel_receive_unittest",
-      "../api:array_view",
-      "../api:frame_transformer_interface",
       "../api:libjingle_peerconnection_api",
-      "../api:make_ref_counted",
       "../api:mock_audio_mixer",
       "../api:mock_frame_decryptor",
       "../api:mock_frame_encryptor",
       "../api:mock_frame_transformer",
       "../api:mock_transformable_audio_frame",
-      "../api:rtp_headers",
       "../api:scoped_refptr",
       "../api/audio:audio_frame_api",
       "../api/audio:audio_processing_statistics",
@@ -193,7 +189,6 @@ if (rtc_include_tests) {
       "../call:rtp_sender",
       "../common_audio",
       "../logging:mocks",
-      "../modules/audio_coding:audio_coding_module_typedefs",
       "../modules/audio_device:audio_device_impl",  # For TestAudioDeviceModule
       "../modules/audio_device:mock_audio_device",
       "../modules/audio_mixer:audio_mixer_impl",
@@ -225,7 +220,6 @@ if (rtc_include_tests) {
       "utility:utility_tests",
       "//testing/gtest",
       "//third_party/abseil-cpp/absl/memory",
-      "//third_party/abseil-cpp/absl/types:optional",
     ]
   }
 
diff --git a/audio/channel_receive_frame_transformer_delegate.cc b/audio/channel_receive_frame_transformer_delegate.cc
index 0b8eacf96e..814127b914 100644
--- a/audio/channel_receive_frame_transformer_delegate.cc
+++ b/audio/channel_receive_frame_transformer_delegate.cc
@@ -10,18 +10,9 @@
 
 #include "audio/channel_receive_frame_transformer_delegate.h"
 
-#include <cstdint>
-#include <memory>
 #include <string>
 #include <utility>
 
-#include "absl/types/optional.h"
-#include "api/array_view.h"
-#include "api/frame_transformer_interface.h"
-#include "api/rtp_headers.h"
-#include "api/scoped_refptr.h"
-#include "api/sequence_checker.h"
-#include "api/task_queue/task_queue_base.h"
 #include "rtc_base/buffer.h"
 
 namespace webrtc {
diff --git a/audio/channel_receive_frame_transformer_delegate.h b/audio/channel_receive_frame_transformer_delegate.h
index 0502458c55..d572ffe8d1 100644
--- a/audio/channel_receive_frame_transformer_delegate.h
+++ b/audio/channel_receive_frame_transformer_delegate.h
@@ -15,7 +15,6 @@
 #include <string>
 
 #include "api/frame_transformer_interface.h"
-#include "api/rtp_headers.h"
 #include "api/sequence_checker.h"
 #include "api/task_queue/task_queue_base.h"
 #include "rtc_base/system/no_unique_address.h"
diff --git a/audio/channel_receive_frame_transformer_delegate_unittest.cc b/audio/channel_receive_frame_transformer_delegate_unittest.cc
index 3ab2b49baf..8b819f1a9a 100644
--- a/audio/channel_receive_frame_transformer_delegate_unittest.cc
+++ b/audio/channel_receive_frame_transformer_delegate_unittest.cc
@@ -10,19 +10,12 @@
 
 #include "audio/channel_receive_frame_transformer_delegate.h"
 
-#include <cstdint>
 #include <memory>
 #include <utility>
 
-#include "api/array_view.h"
-#include "api/frame_transformer_interface.h"
-#include "api/make_ref_counted.h"
-#include "api/rtp_headers.h"
-#include "api/scoped_refptr.h"
 #include "api/test/mock_frame_transformer.h"
 #include "api/test/mock_transformable_audio_frame.h"
 #include "audio/channel_send_frame_transformer_delegate.h"
-#include "rtc_base/thread.h"
 #include "test/gmock.h"
 #include "test/gtest.h"
 
diff --git a/audio/channel_send_frame_transformer_delegate_unittest.cc b/audio/channel_send_frame_transformer_delegate_unittest.cc
index 0ed8677ae1..e8b7aef29d 100644
--- a/audio/channel_send_frame_transformer_delegate_unittest.cc
+++ b/audio/channel_send_frame_transformer_delegate_unittest.cc
@@ -10,20 +10,13 @@
 
 #include "audio/channel_send_frame_transformer_delegate.h"
 
-#include <cstdint>
 #include <memory>
 #include <utility>
 #include <vector>
 
 #include "absl/memory/memory.h"
-#include "absl/types/optional.h"
-#include "api/array_view.h"
-#include "api/frame_transformer_interface.h"
-#include "api/make_ref_counted.h"
-#include "api/scoped_refptr.h"
 #include "api/test/mock_frame_transformer.h"
 #include "api/test/mock_transformable_audio_frame.h"
-#include "modules/audio_coding/include/audio_coding_module_typedefs.h"
 #include "rtc_base/task_queue_for_test.h"
 #include "test/gmock.h"
 #include "test/gtest.h"
diff --git a/media/BUILD.gn b/media/BUILD.gn
index 8b5fd93338..6244b6b4f0 100644
--- a/media/BUILD.gn
+++ b/media/BUILD.gn
@@ -358,7 +358,6 @@ rtc_library("codec") {
     ":media_constants",
     "../api:rtp_parameters",
     "../api/audio_codecs:audio_codecs_api",
-    "../api/video_codecs:scalability_mode",
     "../api/video_codecs:video_codecs_api",
     "../rtc_base:checks",
     "../rtc_base:logging",
diff --git a/media/base/codec.cc b/media/base/codec.cc
index 820cb97980..2e7cc6ac8c 100644
--- a/media/base/codec.cc
+++ b/media/base/codec.cc
@@ -10,21 +10,11 @@
 
 #include "media/base/codec.h"
 
-#include <algorithm>
-#include <cstddef>
-#include <iterator>
-#include <string>
-#include <vector>
-
 #include "absl/algorithm/container.h"
 #include "absl/strings/match.h"
-#include "absl/types/optional.h"
 #include "api/audio_codecs/audio_format.h"
-#include "api/media_types.h"
-#include "api/rtp_parameters.h"
 #include "api/video_codecs/av1_profile.h"
 #include "api/video_codecs/h264_profile_level_id.h"
-#include "api/video_codecs/sdp_video_format.h"
 #ifdef RTC_ENABLE_H265
 #include "api/video_codecs/h265_profile_tier_level.h"
 #endif
diff --git a/media/base/codec.h b/media/base/codec.h
index 5230779635..99a6852462 100644
--- a/media/base/codec.h
+++ b/media/base/codec.h
@@ -11,7 +11,8 @@
 #ifndef MEDIA_BASE_CODEC_H_
 #define MEDIA_BASE_CODEC_H_
 
-#include <cstddef>
+#include <map>
+#include <set>
 #include <string>
 #include <vector>
 
@@ -20,7 +21,6 @@
 #include "absl/types/optional.h"
 #include "api/audio_codecs/audio_format.h"
 #include "api/rtp_parameters.h"
-#include "api/video_codecs/scalability_mode.h"
 #include "api/video_codecs/sdp_video_format.h"
 #include "media/base/media_constants.h"
 #include "rtc_base/system/rtc_export.h"
diff --git a/modules/rtp_rtcp/BUILD.gn b/modules/rtp_rtcp/BUILD.gn
index 58317c1291..0c878a5c16 100644
--- a/modules/rtp_rtcp/BUILD.gn
+++ b/modules/rtp_rtcp/BUILD.gn
@@ -308,7 +308,6 @@ rtc_library("rtp_rtcp") {
     "../../api/video:video_frame_type",
     "../../api/video:video_layers_allocation",
     "../../api/video:video_rtp_headers",
-    "../../api/video_codecs:video_codecs_api",
     "../../call:rtp_interfaces",
     "../../call:video_stream_api",
     "../../common_video",
diff --git a/modules/rtp_rtcp/source/rtp_sender_video_frame_transformer_delegate.cc b/modules/rtp_rtcp/source/rtp_sender_video_frame_transformer_delegate.cc
index 8c24160f38..ae80f7678f 100644
--- a/modules/rtp_rtcp/source/rtp_sender_video_frame_transformer_delegate.cc
+++ b/modules/rtp_rtcp/source/rtp_sender_video_frame_transformer_delegate.cc
@@ -10,31 +10,15 @@
 
 #include "modules/rtp_rtcp/source/rtp_sender_video_frame_transformer_delegate.h"
 
-#include <cstddef>
-#include <cstdint>
-#include <memory>
 #include <string>
 #include <utility>
 #include <vector>
 
-#include "absl/types/optional.h"
-#include "api/array_view.h"
-#include "api/frame_transformer_interface.h"
-#include "api/scoped_refptr.h"
 #include "api/sequence_checker.h"
 #include "api/task_queue/task_queue_factory.h"
-#include "api/transport/rtp/dependency_descriptor.h"
-#include "api/units/time_delta.h"
-#include "api/units/timestamp.h"
-#include "api/video/encoded_image.h"
-#include "api/video/video_codec_type.h"
-#include "api/video/video_frame_metadata.h"
-#include "api/video/video_frame_type.h"
-#include "api/video/video_layers_allocation.h"
-#include "api/video_codecs/video_codec.h"
-#include "modules/rtp_rtcp/source/rtp_video_header.h"
+#include "modules/rtp_rtcp/source/rtp_descriptor_authentication.h"
 #include "rtc_base/checks.h"
-#include "rtc_base/synchronization/mutex.h"
+#include "rtc_base/logging.h"
 
 namespace webrtc {
 namespace {
diff --git a/modules/rtp_rtcp/source/rtp_sender_video_frame_transformer_delegate.h b/modules/rtp_rtcp/source/rtp_sender_video_frame_transformer_delegate.h
index a32c7084ea..243f22ca9f 100644
--- a/modules/rtp_rtcp/source/rtp_sender_video_frame_transformer_delegate.h
+++ b/modules/rtp_rtcp/source/rtp_sender_video_frame_transformer_delegate.h
@@ -21,9 +21,7 @@
 #include "api/task_queue/task_queue_factory.h"
 #include "api/units/time_delta.h"
 #include "api/units/timestamp.h"
-#include "api/video/encoded_image.h"
 #include "api/video/video_layers_allocation.h"
-#include "modules/rtp_rtcp/source/rtp_video_header.h"
 #include "rtc_base/synchronization/mutex.h"
 
 namespace webrtc {
diff --git a/pc/BUILD.gn b/pc/BUILD.gn
index 25dd280a8a..d1b6b297da 100644
--- a/pc/BUILD.gn
+++ b/pc/BUILD.gn
@@ -451,15 +451,7 @@ rtc_source_set("rtp_receiver_proxy") {
   sources = [ "rtp_receiver_proxy.h" ]
   deps = [
     ":proxy",
-    "../api:dtls_transport_interface",
-    "../api:frame_transformer_interface",
     "../api:libjingle_peerconnection_api",
-    "../api:media_stream_interface",
-    "../api:rtp_parameters",
-    "../api:scoped_refptr",
-    "../api/crypto:frame_decryptor_interface",
-    "../api/transport/rtp:rtp_source",
-    "//third_party/abseil-cpp/absl/types:optional",
   ]
 }
 
@@ -996,7 +988,6 @@ rtc_source_set("sdp_offer_answer") {
     "../api:dtls_transport_interface",
     "../api:field_trials_view",
     "../api:libjingle_peerconnection_api",
-    "../api:make_ref_counted",
     "../api:media_stream_interface",
     "../api:rtc_error",
     "../api:rtp_parameters",
@@ -1007,10 +998,8 @@ rtc_source_set("sdp_offer_answer") {
     "../api/crypto:options",
     "../api/video:builtin_video_bitrate_allocator_factory",
     "../api/video:video_bitrate_allocator_factory",
-    "../api/video:video_codec_constants",
     "../media:codec",
     "../media:media_channel",
-    "../media:media_engine",
     "../media:rid_description",
     "../media:stream_params",
     "../p2p:connection",
@@ -1528,7 +1517,6 @@ rtc_library("rtp_transceiver") {
     "../api/task_queue",
     "../api/task_queue:pending_task_safety_flag",
     "../api/video:video_bitrate_allocator_factory",
-    "../api/video_codecs:scalability_mode",
     "../media:codec",
     "../media:media_channel",
     "../media:media_channel_impl",
@@ -2411,7 +2399,6 @@ if (rtc_include_tests && !build_with_chromium) {
       "../test:run_loop",
       "../test:scoped_key_value_config",
       "../test/pc/sctp:fake_sctp_transport",
-      "//testing/gtest",
       "//third_party/abseil-cpp/absl/algorithm:container",
       "//third_party/abseil-cpp/absl/memory",
       "//third_party/abseil-cpp/absl/strings",
diff --git a/pc/rtp_receiver.cc b/pc/rtp_receiver.cc
index ce221a266d..9431aab10a 100644
--- a/pc/rtp_receiver.cc
+++ b/pc/rtp_receiver.cc
@@ -13,12 +13,9 @@
 #include <stddef.h>
 
 #include <atomic>
-#include <string>
 #include <utility>
 #include <vector>
 
-#include "api/media_stream_interface.h"
-#include "api/scoped_refptr.h"
 #include "pc/media_stream.h"
 #include "pc/media_stream_proxy.h"
 #include "rtc_base/thread.h"
diff --git a/pc/rtp_receiver.h b/pc/rtp_receiver.h
index 53451a3333..16ab011f14 100644
--- a/pc/rtp_receiver.h
+++ b/pc/rtp_receiver.h
@@ -21,11 +21,20 @@
 #include <vector>
 
 #include "absl/types/optional.h"
+#include "api/crypto/frame_decryptor_interface.h"
 #include "api/dtls_transport_interface.h"
 #include "api/media_stream_interface.h"
+#include "api/media_types.h"
+#include "api/rtp_parameters.h"
 #include "api/rtp_receiver_interface.h"
 #include "api/scoped_refptr.h"
+#include "api/video/video_frame.h"
+#include "api/video/video_sink_interface.h"
+#include "api/video/video_source_interface.h"
 #include "media/base/media_channel.h"
+#include "media/base/video_broadcaster.h"
+#include "pc/video_track_source.h"
+#include "rtc_base/thread.h"
 
 namespace webrtc {
 
diff --git a/pc/rtp_receiver_proxy.h b/pc/rtp_receiver_proxy.h
index 0cdf6ff528..db677d818a 100644
--- a/pc/rtp_receiver_proxy.h
+++ b/pc/rtp_receiver_proxy.h
@@ -14,16 +14,7 @@
 #include <string>
 #include <vector>
 
-#include "absl/types/optional.h"
-#include "api/crypto/frame_decryptor_interface.h"
-#include "api/dtls_transport_interface.h"
-#include "api/frame_transformer_interface.h"
-#include "api/media_stream_interface.h"
-#include "api/media_types.h"
-#include "api/rtp_parameters.h"
 #include "api/rtp_receiver_interface.h"
-#include "api/scoped_refptr.h"
-#include "api/transport/rtp/rtp_source.h"
 #include "pc/proxy.h"
 
 namespace webrtc {
diff --git a/pc/rtp_sender_receiver_unittest.cc b/pc/rtp_sender_receiver_unittest.cc
index b73edf5ca9..c6c342e940 100644
--- a/pc/rtp_sender_receiver_unittest.cc
+++ b/pc/rtp_sender_receiver_unittest.cc
@@ -18,15 +18,14 @@
 #include <vector>
 
 #include "absl/algorithm/container.h"
+#include "absl/memory/memory.h"
 #include "absl/types/optional.h"
-#include "api/audio_codecs/audio_codec_pair_id.h"
 #include "api/audio_options.h"
 #include "api/crypto/crypto_options.h"
 #include "api/crypto/frame_decryptor_interface.h"
 #include "api/crypto/frame_encryptor_interface.h"
 #include "api/dtmf_sender_interface.h"
 #include "api/environment/environment_factory.h"
-#include "api/make_ref_counted.h"
 #include "api/media_stream_interface.h"
 #include "api/rtc_error.h"
 #include "api/rtc_event_log/rtc_event_log.h"
@@ -52,6 +51,7 @@
 #include "p2p/base/p2p_constants.h"
 #include "pc/audio_rtp_receiver.h"
 #include "pc/audio_track.h"
+#include "pc/channel.h"
 #include "pc/dtls_srtp_transport.h"
 #include "pc/local_audio_source.h"
 #include "pc/media_stream.h"
@@ -63,7 +63,6 @@
 #include "rtc_base/checks.h"
 #include "rtc_base/gunit.h"
 #include "rtc_base/thread.h"
-#include "rtc_base/unique_id_generator.h"
 #include "test/gmock.h"
 #include "test/gtest.h"
 #include "test/run_loop.h"
diff --git a/pc/rtp_transceiver.cc b/pc/rtp_transceiver.cc
index f0a09ce1a5..e585667843 100644
--- a/pc/rtp_transceiver.cc
+++ b/pc/rtp_transceiver.cc
@@ -12,50 +12,25 @@
 
 #include <stdint.h>
 
-#include <algorithm>
-#include <cstddef>
-#include <functional>
 #include <iterator>
-#include <memory>
 #include <set>
 #include <string>
 #include <utility>
 #include <vector>
 
 #include "absl/algorithm/container.h"
-#include "absl/strings/string_view.h"
-#include "absl/types/optional.h"
-#include "api/array_view.h"
+#include "absl/memory/memory.h"
 #include "api/audio_codecs/audio_codec_pair_id.h"
-#include "api/audio_options.h"
-#include "api/crypto/crypto_options.h"
 #include "api/field_trials_view.h"
-#include "api/jsep.h"
-#include "api/media_types.h"
-#include "api/rtc_error.h"
 #include "api/rtp_parameters.h"
-#include "api/rtp_receiver_interface.h"
-#include "api/rtp_sender_interface.h"
-#include "api/rtp_transceiver_direction.h"
-#include "api/scoped_refptr.h"
 #include "api/sequence_checker.h"
-#include "api/task_queue/pending_task_safety_flag.h"
-#include "api/task_queue/task_queue_base.h"
-#include "api/video/video_bitrate_allocator_factory.h"
-#include "api/video_codecs/scalability_mode.h"
 #include "media/base/codec.h"
 #include "media/base/media_channel.h"
-#include "media/base/media_config.h"
+#include "media/base/media_channel_impl.h"
+#include "media/base/media_constants.h"
 #include "media/base/media_engine.h"
 #include "pc/channel.h"
-#include "pc/channel_interface.h"
-#include "pc/connection_context.h"
 #include "pc/rtp_media_utils.h"
-#include "pc/rtp_receiver.h"
-#include "pc/rtp_receiver_proxy.h"
-#include "pc/rtp_sender.h"
-#include "pc/rtp_sender_proxy.h"
-#include "pc/rtp_transport_internal.h"
 #include "pc/session_description.h"
 #include "rtc_base/checks.h"
 #include "rtc_base/logging.h"
diff --git a/pc/sdp_offer_answer.cc b/pc/sdp_offer_answer.cc
index 8b348ba1cf..68d89e439f 100644
--- a/pc/sdp_offer_answer.cc
+++ b/pc/sdp_offer_answer.cc
@@ -12,90 +12,56 @@
 
 #include <algorithm>
 #include <cstddef>
-#include <cstdint>
-#include <functional>
 #include <iterator>
 #include <map>
 #include <memory>
 #include <queue>
-#include <set>
 #include <string>
 #include <utility>
-#include <vector>
 
 #include "absl/algorithm/container.h"
 #include "absl/memory/memory.h"
 #include "absl/strings/match.h"
 #include "absl/strings/string_view.h"
-#include "absl/types/optional.h"
 #include "api/array_view.h"
-#include "api/candidate.h"
 #include "api/crypto/crypto_options.h"
-#include "api/jsep.h"
-#include "api/jsep_ice_candidate.h"
-#include "api/make_ref_counted.h"
-#include "api/media_stream_interface.h"
-#include "api/media_types.h"
-#include "api/peer_connection_interface.h"
-#include "api/rtc_error.h"
+#include "api/dtls_transport_interface.h"
+#include "api/field_trials_view.h"
 #include "api/rtp_parameters.h"
 #include "api/rtp_receiver_interface.h"
 #include "api/rtp_sender_interface.h"
-#include "api/rtp_transceiver_direction.h"
-#include "api/rtp_transceiver_interface.h"
-#include "api/scoped_refptr.h"
-#include "api/sequence_checker.h"
-#include "api/set_local_description_observer_interface.h"
-#include "api/set_remote_description_observer_interface.h"
-#include "api/uma_metrics.h"
 #include "api/video/builtin_video_bitrate_allocator_factory.h"
-#include "api/video/video_codec_constants.h"
 #include "media/base/codec.h"
-#include "media/base/media_engine.h"
 #include "media/base/rid_description.h"
-#include "media/base/stream_params.h"
 #include "p2p/base/ice_transport_internal.h"
 #include "p2p/base/p2p_constants.h"
 #include "p2p/base/p2p_transport_channel.h"
-#include "p2p/base/port_allocator.h"
+#include "p2p/base/port.h"
 #include "p2p/base/transport_description.h"
 #include "p2p/base/transport_description_factory.h"
 #include "p2p/base/transport_info.h"
 #include "pc/channel_interface.h"
-#include "pc/connection_context.h"
 #include "pc/dtls_transport.h"
-#include "pc/jsep_transport_controller.h"
 #include "pc/legacy_stats_collector.h"
-#include "pc/media_session.h"
 #include "pc/media_stream.h"
-#include "pc/media_stream_observer.h"
 #include "pc/media_stream_proxy.h"
 #include "pc/peer_connection_internal.h"
 #include "pc/peer_connection_message_handler.h"
 #include "pc/rtp_media_utils.h"
-#include "pc/rtp_receiver.h"
+#include "pc/rtp_receiver_proxy.h"
 #include "pc/rtp_sender.h"
 #include "pc/rtp_sender_proxy.h"
-#include "pc/rtp_transceiver.h"
-#include "pc/rtp_transmission_manager.h"
-#include "pc/session_description.h"
 #include "pc/simulcast_description.h"
-#include "pc/stream_collection.h"
-#include "pc/transceiver_list.h"
 #include "pc/usage_pattern.h"
 #include "pc/used_ids.h"
 #include "pc/webrtc_session_description_factory.h"
-#include "rtc_base/checks.h"
 #include "rtc_base/helpers.h"
 #include "rtc_base/logging.h"
-#include "rtc_base/operations_chain.h"
 #include "rtc_base/rtc_certificate.h"
 #include "rtc_base/ssl_stream_adapter.h"
 #include "rtc_base/string_encode.h"
 #include "rtc_base/strings/string_builder.h"
-#include "rtc_base/thread.h"
 #include "rtc_base/trace_event.h"
-#include "rtc_base/weak_ptr.h"
 #include "system_wrappers/include/metrics.h"
 
 using cricket::ContentInfo;
diff --git a/pc/sdp_offer_answer.h b/pc/sdp_offer_answer.h
index 8b845f46ef..88ddfe0332 100644
--- a/pc/sdp_offer_answer.h
+++ b/pc/sdp_offer_answer.h
@@ -55,6 +55,7 @@
 #include "pc/stream_collection.h"
 #include "pc/transceiver_list.h"
 #include "pc/webrtc_session_description_factory.h"
+#include "rtc_base/checks.h"
 #include "rtc_base/operations_chain.h"
 #include "rtc_base/ssl_stream_adapter.h"
 #include "rtc_base/thread.h"
